component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java

```diff
@@ -1115,16 +1117,41 @@ public void onCreate(final Container container) {
             try {
                 String alreadyScannedClasses = null;
                 Filter filter = KnownClassesFilter.INSTANCE;
-                try (final InputStream containerFilterConfig =
-                        container.getLoader().getResourceAsStream("TALEND-INF/scanning.properties")) {
-                    if (containerFilterConfig != null) {
+                // we need to scan the nested repository
+                if (container.hasNestedRepository()) {
+                    try {
+                        final Enumeration<URL> urls = loader.getResources("TALEND-INF/scanning.properties");
                         final Properties config = new Properties();
-                        config.load(containerFilterConfig);
-                        filter = createScanningFilter(config);
-                        alreadyScannedClasses = config.getProperty("classes.list");
+                        while (urls.hasMoreElements()) {
+                            final URL url = urls.nextElement();
+                            // ensure we scan the correct classes of plugin in the nested repository
+                            if ("nested".equals(url.getProtocol())
+                                    && url.getPath().contains(container.getRootModule())) {
+                                try (final BufferedReader reader =
+                                        new BufferedReader(new InputStreamReader(url.openStream()))) {
+                                    config.load(reader);
+                                    filter = createScanningFilter(config);
+                                    alreadyScannedClasses = config.getProperty("classes.list");
```

Comment:
should we "break" when we find the nested resource and create "filter" and "alreadyScannedClasses"?
because otherwise in case if we find more, the previous values will be overwritten
should we accumulate it maybe?

Reply:
We could add a break statement to stop iteration.
Anyway, normally we won't find any other resource as we check `url.getPath().contains(container.getRootModule())`